Brendan, thank you for the report and for the patches.

The definition HAVE_STRING_H is supposed to be generated by the configure script, in config.h which is then included by udint.h. In standalone mode, one could manually create their own config.h file, though not strictly necessary.

I'm not sure I understand why there is an error message. Standard C dictates that undefined identifiers in "#if" preprocessor directives evaluate to 0. So "#if HAVE_STRING_H" is an acceptable use of the directive. For instance,

```
$ cat test.c
#if NOT_DEFINED + 1 == 1
# warning I was evaluated
#endif 

$ gcc -E test.c
# 1 "test.c"
# 1 "<built-in>"
# 1 "<command-line>"
# 1 "test.c"
test.c:2:3: warning: #warning I was evaluated

$ gcc -v
Target: i686-apple-darwin11
gcc version 4.2.1 (Based on Apple Inc. build 5658) (LLVM build 2336.11.00)
```

There may something about the webkit build environment and/or the copy of udis86 you are using, perhaps? Let me know what you think.

The other changes look good. Thanks.

--- a/libudis86/extern.h
+++ b/libudis86/extern.h
@@ -78,7 +78,7 @@ extern const struct ud_operand* ud_insn_opr(const struct ud *u, unsigned int n);
 
 extern int ud_opr_is_sreg(const struct ud_operand *opr);
 
-extern int ud_opr_isgpr(const struct ud_operand *opr);
+extern int ud_opr_is_gpr(const struct ud_operand *opr);
 
 extern const char* ud_lookup_mnemonic(enum ud_mnemonic_code c);
 

--- a/libudis86/syn.c
+++ b/libudis86/syn.c
@@ -104,7 +104,7 @@ ud_syn_rel_target(struct ud *u, struct ud_operand *opr)
  *    returns a negative number and truncates the output.
  */
 int
-ud_asmprintf(struct ud *u, char *fmt, ...)
+ud_asmprintf(struct ud *u, const char *fmt, ...)
 {
   int ret;
   int avail;

--- a/libudis86/syn.h
+++ b/libudis86/syn.h
@@ -36,10 +36,10 @@ extern const char* ud_reg_tab[];
 uint64_t ud_syn_rel_target(struct ud*, struct ud_operand*);
 
 #ifdef __GNUC__
-int ud_asmprintf(struct ud *u, char *fmt, ...) 
+int ud_asmprintf(struct ud *u, const char *fmt, ...)
     __attribute__ ((format (printf, 2, 3)));
 #else
-int ud_asmprintf(struct ud *u, char *fmt, ...);
+int ud_asmprintf(struct ud *u, const char *fmt, ...);
 #endif
 
 void ud_syn_print_addr(struct ud *u, uint64_t addr);

